src/meta/src/controller/streaming_job.rs

```diff
@@ -1656,6 +1659,137 @@ impl CatalogController {
             .await
     }
 
+    pub async fn update_sink_props_by_sink_id(
+        &self,
+        sink_id: SinkId,
+        props: BTreeMap<String, String>,
+    ) -> MetaResult<HashMap<String, String>> {
+        let inner = self.inner.read().await;
+        let txn = inner.db.begin().await?;
+
+        let (sink, _obj) = Sink::find_by_id(sink_id)
+            .find_also_related(Object)
+            .one(&txn)
+            .await?
+            .ok_or_else(|| MetaError::catalog_id_not_found(ObjectType::Sink.as_str(), sink_id))?;
+
+        // Validate that props can be altered
+        match sink.properties.inner_ref().get(CONNECTOR_TYPE_KEY) {
+            Some(connector) => {
+                let connector_type = connector.to_lowercase();
+                match_sink_name_str!(
+                    connector_type.as_str(),
+                    SinkType,
+                    {
+                        for (k, v) in &props {
+                            if !SinkType::SINK_ALTER_CONFIG_LIST.contains(&k.as_str()) {
+                                return Err(SinkError::Config(anyhow!(
+                                    "unsupported alter config: {}={}",
+                                    k,
+                                    v
+                                ))
+                                .into());
+                            }
+                        }
+                        SinkType::validate_alter_config(&props)
+                    },
+                    |sink: &str| Err(SinkError::Config(anyhow!("unsupported sink type {}", sink)))
+                )?
+            }
+            None => {
+                return Err(
+                    SinkError::Config(anyhow!("connector not specified when alter sink")).into(),
+                );
+            }
+        };
+
+        let mut new_config = sink.properties.clone().into_inner();
+        new_config.extend(props);
+
+        let definition = sink.definition.clone();
+        let entries: &Vec<String> = &new_config
+            .iter()
+            .map(|(k, v)| format!("{} = {}", k, v))
+            .collect();
+        let new_with_definition = format!("WITH ( {} )", entries.join(", "));
+        let re = Regex::new(r"(?i)WITH\s?\([^\)]*\)").unwrap();
+        let new_definition = re.replace_all(&definition, new_with_definition);
```

Comment:
shall we re-generate the sql here using `CreateSinkStatement`, instead of magic.